### ISSUE TITLE
Disabled ipAddress tracking option doesn't reflect on IP address of a user on amplitude.com

### ISSUE THREAD

**Issue by alok1111** [amplitude/Amplitude-TypeScript#492]
## Expected Behavior
IP address of a user doesn't appear in `User Look-Up > some_user > Event Stream > some_event > Other > IP Address`

## Current Behavior
IP address appears

## Possible Solution
-

## Steps to Reproduce
1. Initialize SDK with `ipAddress` tracking option disabled
2. Execute `track` method add see ip address on the amplitude.com

Sample code:
```
        init(apiKey, userId, {
            logLevel: Types.LogLevel.None,
            defaultTracking: {
                attribution: false,
                pageViews: false,
                sessions: true,
                formInteractions: false,
                fileDownloads: false
            },
            trackingOptions: {
                ipAddress: false,
                language: true,
                platform: true
            }
        })

        track(actionName, {
            elementName
        })
```

## Environment
- JS SDK Version: 2.1.2
- Installation Method: Yarn
- Browser and Version: Firefox 115

**Comment by Mercy811** [comment]
Hi @alok1111, have you ever sent IP before disabling it? If so, Amplitude's backend uses history IP data. Could you log the payload of events sent to see if IP and location are include (or share your project link so I can check raw requests)? If not, it means the SDK works as expected. You can:

- simply test with another user id
- create another project and send events to the new project
- clear previous properties by 
```
amplitude.identify(
  new amplitude.Identify().identify.clearAll()
)
```

**Comment by CoryMortimer** [comment]
@Mercy811 I believe I am running into this issue as well. I was looking through the source and I found [this section of code](https://github.com/amplitude/Amplitude-TypeScript/blob/c6355eff3f8e366bee43b3af5f0d2c57d90941d5/packages/analytics-browser/src/config.ts#L215-L220) that may be the culprit. It appears that any option passed to `options.trackingOptions` will be forced to `true`.

I do not know the code base that well, so in case I'm wrong, I ran your testing steps as well.

I am using ampli instead of directly interfacing with the amplitude SDK. Here are my steps:
```typescript
ampli.load({
  client: {
    configuration: {
      identityStorage: 'none',
      serverUrl: '/custom-endpoint',
      trackingOptions: {
        ipAddress: false
      }
    }
  }
});
```
I then triggered an event.
At `/custom-endpoint`, I added a log statement to see what the body contained. Here's a small snippet:
```typescript
{
  "api_key": "...",
  "events": [
    {
      ...
      // In your above comment, you mentioned "location". I don't see
      // that key in this payload but I do see the following "ip" key:
      "ip": "$remote",
      ...
      "library": "amplitude-ts/2.1.2",
      ...
    }
  ],
  "options": {}
}
```

I used a different user id, triggered an event, and still saw the ip address attached to the new user id on the user page.

**Comment by alok1111** [comment]
> have you ever sent IP before disabling it?
I did but for other user IDs.

> Could you log the payload of events sent to see if IP and location are include (or share your project link so I can check raw requests)? If not, it means the SDK works as expected.

This the body of the POST `https://api2.amplitude.com/2/httpapi` (API key is modified)
```
{
  "api_key": "1ea1a00d0494ea8b6d2cxxxxxxxxxxxx",
  "events": [
    {
      "user_id": "9dd0097384f5f9a4c9b8c31e2ea79ae4",
      "device_id": "c30da266-4ca6-4ec6-822c-4f4575510731",
      "session_id": 1690280643815,
      "time": 1690280643815,
      "platform": "Web",
      "language": "en-US",
      "ip": "$remote",
      "insert_id": "0e68ac4f-ef77-4e30-8557-dc5a75f9257b",
      "event_type": "session_start",
      "event_id": 0,
      "library": "amplitude-ts/2.1.2",
      "user_agent": "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0"
    }
  ],
  "options": {}
}
```
It's probably not an SDK issue, but I believe it's an Amplitude ecosystem issue. Please let me know whom I should reach to solve it. It's a huge blocker for us to start using Amplitude.

**Comment by Mercy811** [comment]
Hi @alok1111, I just created a PR to fix. It should be merged soon. I will let you know once it's merged. Please upgrade to the latest version after that.

**Comment by Mercy811** [comment]
Hi @alok1111 @CoryMortimer, it's fixed on `@amplitude/analytics-browser@2.1.3`. Please upgrade to the newest version.

**Comment by alok1111** [comment]
Hi @Mercy811, I can confirm the IP address doesn't appear in the event description on amplitude.com. Thank you for the fast fix!
